File: 2 - Simple Linear Regression/simple_linear_regression.py
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Importing datasets
dataset = pd.read_csv("Salary_Data.csv")
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#Splitting in training and testing sets
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=1/3, random_state=0)

#Feature Scaling is done while fitting the model
#==============================================================================
# from sklearn.preprocessing import StandardScaler
# sc_X = StandardScaler()
# X_train = sc_X.fit_transform(X_train)
# X_test = sc_X.transform(X_test) #already fitted on training set
#==============================================================================

#Fitting model to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train, y_train)

#Predicting test results
y_pred = regressor.predict(X_test)

#Visualizing the results - Training Set
plt.scatter(X_train, y_train, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary x Experience (Training Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Annual Salary')
plt.show()

#Visualizing the results - Testing Set
plt.scatter(X_test, y_test, color='red')
plt.plot(X_train, regressor.predict(X_train), color='blue')
plt.title('Salary x Experience (Testing Set)')
plt.xlabel('Years of Experience')
plt.ylabel('Annual Salary')
plt.show()

File: 2 - Simple Linear Regression/simple_linear_regression.R
# Importing the dataset
dataset = read.csv(file = "Salary_Data.csv")
#dataset = dataset[, 2:3] #

# Splitting the dataset in train and test set
library(caTools)
set.seed(123)
split = sample.split(Y = dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
testing_set = subset(x = dataset, subset = split == FALSE)

# Fitting linear regression model to the training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

# Predicting test set results
y_pred = predict(regressor, newdata = testing_set)

# Visualizing training set results
library(ggplot2)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y =  predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary x  Experience - Training set') +
  xlab('Years of Experience') + 
  ylab('Annual Salary')

# Visualizing testing set results
ggplot() + 
  geom_point(aes(x = testing_set$YearsExperience, y = testing_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y =  predict(regressor, newdata = training_set)),
            colour = 'blue') + 
  ggtitle('Salary x  Experience - Testing set') +
  xlab('Years of Experience') + 
  ylab('Annual Salary')